profiling/roctx-connector/kp_roctx_connector.cpp

```diff
@@ -29,29 +31,40 @@ struct Section {
 std::vector<Section> kokkosp_sections;
 }  // namespace
 
-struct Kokkos_Tools_ToolSettings {
-  bool requires_global_fencing;
-  bool padding[255];
-};
+namespace KokkosTools {
+namespace ROCTXConnector {
 
-extern "C" void kokkosp_request_tool_settings(
+static bool tool_globfences;
+
+void kokkosp_request_tool_settings(
     const uint32_t, Kokkos_Tools_ToolSettings* settings) {
-  settings->requires_global_fencing = false;
+  if (tool_globfences) {
+    settings->requires_global_fencing = true;
+  } else {
+    settings->requires_global_fencing = false;
+  }
 }
 
-extern "C" void kokkosp_init_library(const int loadSeq,
-                                     const uint64_t interfaceVer,
-                                     const uint32_t /*devInfoCount*/,
-                                     void* /*deviceInfo*/) {
+void kokkosp_init_library(const int loadSeq,
+                          const uint64_t interfaceVer,
+                          const uint32_t /*devInfoCount*/,
+                          Kokkos_Profiling_KokkosPDeviceInfo* /*deviceInfo*/) {
+
+  const char* tool_global_fences = std::getenv("KOKKOS_TOOLS_GLOBALFENCES");
+  if (tool_global_fences) {
+    tool_globfences = (atoi(tool_global_fences) != 0);
+  }
+
   std::cout << "-----------------------------------------------------------\n"
             << "KokkosP: ROC Tracer Connector (sequence is " << loadSeq
             << ", version: " << interfaceVer << ")\n"
+            << "Global fences: " << (tool_globfences ? "ON" : "OFF") << "\n"
```

Comment:
I feel like this is something one would only want to see in verbose mode.

Reply:
Indeed. It seems there's currently no "verbose" bode in the different connectors. So, I've reverted the change.